cmd/mc-router/main.go

```diff
@@ -240,6 +266,15 @@ func main() {
 		logrus.WithError(err).Fatal("Unable to start metrics reporter")
 	}
 
+	if config.AutoScale.CacheStatus {
+		logrus.Info("Starting status cache updater")
+		connector.StatusCache.StartUpdater(connector, cacheInterval, func() map[string]string {
+			mappings := server.Routes.GetMappings()
+			logrus.WithField("mappings", mappings).Debug("Updating status cache with mappings")
+			return mappings
+		})
```

Comment:
...and that starting of the cache updater should be encapsulated inside of the `connector.WaitForConnections` method.

As it is, it's leaking a bit too much of runtime behavior into the main method.